bench-exchange/README.md

```diff
@@ -181,46 +184,73 @@ pub enum ExchangeInstruction {
 }
 ```
 
-## Trade requests
+## Order requests
 
 When an investor decides to exchange a token of one type for another, they
-submit a transaction to the Solana Blockchain containing a trade request, which,
-if successful, is turned into a trade order.  Trade orders do not expire but are
-cancellable. <!-- Trade orders should have a timestamp to enable trade
-expiration -->  When a trade order is created, tokens are deducted from a token
-account and the trade order acts as an escrow.  The tokens are held until the
-trade order is fulfilled or canceled. If the direction is `To`, then the number
-of `tokens` are deducted from the primary account, if `From` then `tokens`
-multiplied by `price` are deducted from the secondary account.  Trade orders are
+submit a transaction to the Solana Blockchain containing an order request, which,
+if successfully validated, is turned into an order. Active orders can be cancelled
+by sending a cancel request to the exchange and have an optional expiration field,
+which takes a timestamp after which the order is considered invalid.
+When an order is created, tokens are deducted from a token account and the
+order acts as an escrow.  The tokens are held until the trade order is filled,
+canceled, or expired.
+
+In the event that the order is matched and fully executes, one or more AssetAmounts
+from the offer field will reflect the amount and identity of assets deducted from
+the primary account and credited to the secondary account. orders are
 no longer valid when the number of `tokens` goes to zero, at which point they
-can no longer be used. <!-- Could support refilling trade orders, so trade order
+can no longer be used.
+<!-- Could support refilling trade orders, so trade order
 accounts are refilled rather than accumulating -->
 
+
+
 ```rust
-/// Direction of the exchange between two tokens in a pair
-pub enum Direction {
-    /// Trade first token type (primary) in the pair 'To' the second
-    To,
-    /// Trade first token type in the pair 'From' the second (secondary)
-    From,
+/// holds asset/amount pairs
+pub struct AssetAmount { /// needs better name (AssetPool, AssetInfo ?)
+    /// Asset in question, as identified by contract address
+    pub Asset: Pubkey,
+    /// Amount of the Asset in question in base units
+    pub Amount: u64,
+}
+
+pub enum SettlementType {
+    Auto,     // Automatically settle entire order to underlying asset
+    Deferred, // Transfer tokens but don't begin settlement process
```

Comment:
Would help to define what settlement means in the glossary.  Isn't token transfer part of a settlement process?

Reply:
yeah need to clarify this more, There's really 2 types of settlement we're dealing with here. 
standard token tx - settlement is the transfer of tokens between orders/exchange accounts
compound tx - standard token tx with one or more representation tokens with their own redemption mechanisms

A fully executed cross-chain transfer is really layers of settlement
1 - execute and settle tokens offered/accepted to appropriate accounts
2 - redeem and settle representation tokens to their backing asset on another chain